### PR TITLE
Master prefix replacement and test suite fixes

### PR DESCRIPTION
**Pull request by mingwandroid**

### CODE REVIEW

**Comment by mingwandroid** [comment]
This PR contains cherry-picked commits from my new features WIP branch, but does not contain any new features, only test-suite fixes and a fix for binary prefix replacement on Windows.

For the next release, it is, in my opinion, imperative that at the very least, 
b6f9b43 is used.

File: conda_build/api.py [hunk 1]
```diff
@@ -205,8 +205,8 @@ def build(recipe_paths_or_metadata, post=None, need_source_download=True,
 
     if not absolute_recipes:
         raise ValueError('No valid recipes found for input: {}'.format(recipe_paths_or_metadata))
-    return build_tree(absolute_recipes, config, stats, build_only=build_only, post=post,
-                      notest=notest, need_source_download=need_source_download, variants=variants)
```
Comment (jjhelmus) [inline conda_build/api.py:209]: Was the removal of this parameter intentional?  It is used in the build_tree function.

Comment (mingwandroid) [reply conda_build/api.py:209]: It was unused so I removed it.

Comment (mingwandroid) [reply conda_build/api.py:209]: https://github.com/conda/conda-build/pull/3932/files#diff-ac55d744935b1f37a3369258b88e8f1fL2771-L2772

Comment (jjhelmus) [reply conda_build/api.py:209]: Nevermind, read further and now see that this parameter is not used.

File: conda_build/skeletons/pypi.py [hunk 2]
```diff
@@ -74,11 +75,11 @@
 # etc. is determined by EXPECTED_SECTION_ORDER.
 PYPI_META_STATIC = {
     'package': OrderedDict([
-        ('name', '{{ name|lower }}'),
+        ('name', '{{ namel }}'),
         ('version', '{{ version }}'),
     ]),
     'source': OrderedDict([
-        ('url', '/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz'),  # NOQA
+        ('url', '/packages/source/{{ namel[0] }}/{{ namel }}/{{ namel }}-{{ version }}.tar.gz'),  # NOQA
```
Comment (jjhelmus) [inline conda_build/skeletons/pypi.py:82]: The PyPI is case sensitive to the package name.

For example the following will give a 404:
https://pypi.io/packages/source/f/flask/flask-1.1.1.tar.gz
The correct URL is:
https://pypi.io/packages/source/F/Flask/Flask-1.1.1.tar.gz

**Comment by mingwandroid** [comment]
Thanks for the review Jonathan, I'll wait for the latest builds to go through in-case anything comes up.

File: tests/test_published_examples.py [hunk 3]
```diff
@@ -15,7 +15,8 @@
 def test_skeleton_pypi(testing_workdir):
     """published in docs at http://conda.pydata.org/docs/build_tutorials/pkgs.html"""
     conda_path = os.path.join(sys.prefix, 'Scripts' if sys.platform == 'win32' else 'bin', 'conda')
-    cmd = conda_path + ' skeleton pypi Click'
+    # Note, if you use Click here instead it will fail.
+    cmd = conda_path + ' skeleton pypi Flask'
     check_call_env(cmd.split())
     cmd = conda_path + ' build click'
```
Comment (jjhelmus) [inline tests/test_published_examples.py:21]:
```suggestion
    cmd = conda_path + ' build flask'
```

File: tests/test_published_examples.py [hunk 4]
```diff
@@ -15,7 +15,8 @@
 def test_skeleton_pypi(testing_workdir):
     """published in docs at http://conda.pydata.org/docs/build_tutorials/pkgs.html"""
     conda_path = os.path.join(sys.prefix, 'Scripts' if sys.platform == 'win32' else 'bin', 'conda')
-    cmd = conda_path + ' skeleton pypi Click'
+    # Note, if you use Click here instead it will fail.
+    cmd = conda_path + ' skeleton pypi Flask'
```
Comment (jjhelmus) [inline tests/test_published_examples.py:19]:
```suggestion
    cmd = conda_path + ' skeleton pypi click'
```

File: tests/test_published_examples.py (hunk 3 above)
Comment (jjhelmus) [reply tests/test_published_examples.py:21]: Either this or the other, not both.  

PyPI, your package name normalization needs some work.

**Pull request merged**